src/components/SetupTab/index.jsx

```diff
@@ -2,41 +2,354 @@ import React from 'react';
+    const payload = { 
+      filepath: filepath,
+      modelname: this.props.modelName,
+      pyname: this.props.pyModuleName,
+      args: args_dict_string
+    }
+    saveToPython(payload);
+  }
+
+  wrapArgsToJsonFile(datastackPath) {
+    this.props.argsToJsonFile(datastackPath, this.state.argsValues)
+  }
+
+  wrapInvestExecute() {
+    this.props.investExecute(this.state.argsValues)
+  }
+
+  updateArgValues(key, value) {
+    /** Update state with arg values as they change, and validate the args. 
+    *
+    * Updating means 
+    * 1) setting the value
+    * 2) 'touching' the arg - implications for display of validation warnings
+    * 3) toggling the enabled/disabled/hidden state of any dependent args
+    *
+    * @param {string} key - the invest argument key
+    * @param {string} value - the invest argument value
+    */
+
+    let argsValues = Object.assign({}, this.state.argsValues);
+    argsValues[key]['value'] = value;
+    argsValues[key]['touched'] = true;
+
+    if (this.props.argsSpec[key].ui_control) {
+      const updatedArgsValues = toggleDependentInputs(
+        this.props.argsSpec, argsValues, key)
+      argsValues = updatedArgsValues;
+    }
+    this.setState({argsValues: argsValues})
+    this.investValidate(argsValues)
+  }
+
+  batchUpdateArgs(argsDict) {
+    /** Update state with values and validate a batch of InVEST arguments.
+    *
+    * @params {object} argsDict - key: value pairs of InVEST arguments.
+    */
+    let { argsValues, argsValidation } = initializeArgValues(
+      this.props.argsSpec, argsDict)
+    Object.keys(this.props.argsSpec).forEach((argkey) => {
+      if (argkey === 'n_workers') { return }
+      const argSpec = Object.assign({}, this.props.argsSpec[argkey])
+      if (argSpec.ui_control) {
+        argsValues = toggleDependentInputs(this.props.argsSpec, argsValues, argkey)
+      }
+    })
+    
+    this.setState({
+      argsValues: argsValues,
+      argsValidation: argsValidation},
+      () => this.investValidate(this.state.argsValues)
+    )
+  }
+
+  async investValidate(argsValues, limit_to) {
+    /** Validate an arguments dictionary using the InVEST model's validate function.
+    *
+    * @param {object} args_dict_string - a JSON.stringify'ed object of model argument
+    *    keys and values.
+    * @param {string} limit_to - an argument key if validation should be limited only
+    *    to that argument.
+    */
+    let argsSpec = JSON.parse(JSON.stringify(this.props.argsSpec));
+    let argsValidation = Object.assign({}, this.state.argsValidation);
+    let keyset = new Set(Object.keys(argsSpec));
+    let payload = { 
+      model_module: this.props.pyModuleName,
+      args: argsDictFromObject(argsValues)
+    };
+
+    // TODO: is there a use-case for `limit_to`? 
+    // Right now we're never calling validate with a limit_to,
+    // but we have an awful lot of logic here to cover it.
+    if (limit_to) {
+      payload['limit_to'] = limit_to
+    }

```

Comment:
I think we can safely remove anything that has to do with `limit_to`!  It's still used by the qt interface, but part of the intent of overhauling validation was to remove this optional parameter and all of the complexity that has to be put into place to support it.

Reply:
Sounds good. I made #24 for this. I thought about doing it right here, but it's low priority and I'd rather not trigger a re-review of this PR.